src/python/pants/core/util_rules/environments.py

```diff
@@ -175,6 +178,46 @@ class LocalEnvironmentTarget(Target):
     )
 
 
+class LocalWorkspaceCompatiblePlatformsField(CompatiblePlatformsField):
+    pass
+
+
+class LocalWorkspaceEnvironmentTarget(Target):
+    alias = "workspace_environment"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        LocalWorkspaceCompatiblePlatformsField,
+    )
+    help = help_text(
+        """
+        Configuration of a "workspace" execution environment for specific platforms.
+
+        A "workspace" environment is a local environment which executes build processes within
+        the repository and not in the usual execution sandbox. This is useful when interacting with
+        third-party build orchestration tools which may not run correctly when run from within the Pants
+        execution sandbox.
+
+        SAFETY: Workspace environments are inherently UNSAFE since Pants cannot guarantee that any process invoked
+        as a workspace environment is in fact reproducible. By using this environment, you forgo the
+        "hermetic" nature of the Pants execution model and must guarantee to Pants that the invoked process
+        is repeatable and does not have side effects.
+
+        Environment configuration includes the platforms the environment is compatible with, and
+        optionally a fallback environment, along with environment-aware options (such as
```

Comment:
We've gotten rid of the fallback now right?

Reply:
Yes, I removed the fallback= field from `workspace_environment` target type. Will update these docs.